tiny_dnn/core/framework/tensor.h

```diff
@@ -348,6 +353,22 @@ class Tensor {
         std::copy(shape.begin(), shape.end(), shape_.begin());
     }
 
+    Tensor view_impl(std::initializer_list<size_t> const &start,
+                     std::initializer_list<size_t> const &new_shape) {
+	if (start.size() > kDimensions || new_shape.size() > kDimensions) {
+            throw nn_error("Overpassed number of existing dimensions.");
+	}
+
+	// compute the new offset and check that it's feasible to create
+	// the new view.
+	const size_t new_offset = compute_offset(start, shape_);
+	if (new_offset + product(new_shape) > size_) {
+            throw nn_error("Cannot create a view from this tensor");
+	}
+
+        return Tensor(storage_ptr_, new_offset, new_shape);
```

Comment:
Does this really works if I create a view from a tensor (not a view) and then take the view out of it?
Actually, I think the first view will give bad results already: offset and shape are not enough information to compute correct offsets, you'll also need the original shape.